zb-db/src/main/java/io/camunda/zeebe/db/impl/rocksdb/transaction/ForeignKeyChecker.java

```diff
@@ -30,31 +31,62 @@ public ForeignKeyChecker(
     enabled = settings.enableForeignKeyChecks();
   }
 
-  private void assertExists(
-      final ZeebeTransaction transaction, final DbForeignKey<? extends DbKey> foreignKey)
+  public void assertExists(
+      final ZeebeTransaction transaction, final ContainsForeignKeys containsForeignKey)
       throws Exception {
+    if (!enabled) {
+      return;
+    }
+    for (final var fk : containsForeignKey.containedForeignKeys()) {
+      assertForeignKeyExists(transaction, fk);
+    }
+  }
+
+  private void assertForeignKeyExists(
+      final ZeebeTransaction transaction, final DbForeignKey<DbKey> foreignKey) throws Exception {
+    if (foreignKey.skip().test(foreignKey.inner())) {
```

Comment:
:wrench: this reads a bit strange. I think it would be easier to read if this was a method on `foreignKey`: `if foreignKey.shouldSkip() then return;`